scss/_grid.scss

```diff
@@ -44,18 +44,52 @@
 @if $enable-grid-classes {
   .row {
     @include make-row();
-  }
 
-  // Remove the negative margin from default .row, then the horizontal padding
-  // from all immediate children columns (to prevent runaway style inheritance).
-  .no-gutters {
-    margin-right: 0;
-    margin-left: 0;
+    > * {
+      @include make-col-ready();
+    }
 
-    > .col,
-    > [class*="col-"] {
+    // Remove paddings & margins from nested `.row`s
+    > .row {
       padding-right: 0;
       padding-left: 0;
+      margin-right: 0;
+      margin-left: 0;
+    }
+  }
+}
+
+// Gutters
+//
+// Make use of `.g-*`, `.gx-*` or `.gy-*` utilities to change spacing between the columns.
+
+@if $enable-grid-classes {
+  @each $breakpoint in map-keys($grid-breakpoints) {
+    $infix: breakpoint-infix($breakpoint, $grid-breakpoints);
+
+    @include media-breakpoint-up($breakpoint, $grid-breakpoints) {
+
+      @each $key, $value in $gutters {
+        .g#{$infix}-#{$key},
+        .gx#{$infix}-#{$key} {
+          margin-right: -$value / 2;
+          margin-left: -$value / 2;
+
+          > * {
+            padding-right: $value / 2;
+            padding-left: $value / 2;
+          }
+        }
+
+        .g#{$infix}-#{$key},
+        .gy#{$infix}-#{$key} {
```

Comment:
Should this be `.gb-` since it's only setting `margin-bottom`? And, since it's only `margin-bottom`, does this double up on margin utilities or is it adding meaningful changes/shortcuts?

Reply:
Nope, it adds a bottom margin to the cols, and adds a negative margin to the row, therefore there are only margins between the cols in vertical direction.

You can check out the forms docs to see how that works: https://deploy-preview-28517--twbs-bootstrap.netlify.com/docs/4.3/forms/layout/

Reply:
Rephrasing, `.my-*` is currently used as a shortcut to `margin-top` and `margin-bottom`. In the `.gy-` proposed here, we're only setting `margin-bottom`. It sets the wrong expectation that `.gy-` is affecting only a single vertical `margin` property as opposed to two.

Does that make sense?

Reply:
The `.gy-` classes don't only add margin to the bottom but also counteract the margins of the last cols. So we don't have margins below the cols, but between the cols. The `gb-` class might not be ideal to cover this?

Alternative namings could be:
- `.gap`, `.gapx` & `.gapy`
- `.g`, `.g-inline` & `.g-block` (using the [logical properties](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Logical_Properties))
- `.gap`, `.row-gap` & `.column-gap` (more like the grid [gaps](https://developer.mozilla.org/en-US/docs/Web/CSS/gap))